src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -0,0 +1,186 @@
+        super(OSAOfflineReintegration, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')[0]
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')[0]
+        self.record_length = self.params.get("length", '/run/record/*')
+
+    def get_pool_leader(self):
+        """Get the pool leader
+           Args: None
+           Returns : pool_leader (int)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    def get_pool_version(self):
+        """Get the pool version
+           Args: None
+           Returns : pool_version (int)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][4])
+
+    def write_single_object(self):
+        """Write some data to the existing pool.
+           Args: None
+           Returns : None
+        """
+        self.pool.connect(2)
+        csum = self.params.get("enable_checksum", '/run/container/*')
+        container = DaosContainer(self.context)
+        input_param = container.cont_input_values
+        input_param.enable_chksum = csum
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.log.info("Writing the Single Dataset")
+        record_index = 0
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length[record_index])
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+                record_index = record_index + 1
+                if record_index == len(self.record_length):
+                    record_index = 0
+
+    def run_offline_reintegration_test(self, num_pool, data=False):
+        """Run the offline reintegration without data.
+           Args: num_pool (int) : Total pools to create
+                                  for testing purpose.
+                 data (bool) : False (Pool has no data) -Default
+                               True (Create some data in pool)
+           Returns : None
+        """
```

Comment:
Docstring should be:
```
"""Run the offline reintegration without data.

Args:
    num_pool (int) : total pools to create for testing purposes.
    data (bool) : whether pool has no data or to create some data in pool.
        Defaults to True.
"""
```
If nothing is returned, no need to specify it in the docstring.

Reply:
Done.